src/js/components/accordion.js

```diff
@@ -0,0 +1,270 @@
+      const accordionParent = event.target.closest('[data-accordion]');
+      // If not an accordion, ignore
+      if (!accordionParent) return;
+
+      const currentTrigger = event.target.closest(this.triggerSelector);
+      if (!currentTrigger) return;
+
+      // Create an array of all the focusable elements within the accordion
+      const nextTrigger = this.triggers.indexOf(currentTrigger) + 1;
+      const prevTrigger = this.triggers.indexOf(currentTrigger) - 1;
+
+      switch (event.keyCode) {
+        case keyCodes.up: {
+          if (this.triggers[prevTrigger] === undefined) {
+            this.triggers[this.triggers.length - 1].focus();
+          } else {
+            this.triggers[prevTrigger].focus();
+          }
+          break;
+        }
+
+        case keyCodes.down: {
+          if (this.triggers[nextTrigger] === undefined) {
+            this.triggers[0].focus();
+          } else {
+            this.triggers[nextTrigger].focus();
+          }
+          break;
+        }
+
+        case keyCodes.end: {
+          this.triggers[this.triggers.length - 1].focus();
+          break;
+        }
+
+        case keyCodes.home: {
+          this.triggers[0].focus();
+          break;
+        }
+
+        default: {
+          break;
+        }
+      }
+    }
+  }
+
+  /**
+   * This function is used to initialize accordions on load. It also adds the
+   * appropriate aria-expanded, and hidden attributes to remove some burden
+   * from the develop.
+   */
+  _openOnInit() {
+    // If accordion is set to open all panels on init, open them
+    if (this.element.hasAttribute('data-accordion-init')) {
+      this.panels.forEach((panel) => {
+        this.open(panel);
+      });
+    } else if (this.openOnInit) {
+      // If this.openOnInit has been set, open it on initialization
+      this.open(this.openOnInit);
+      // Keep all other accordions closed by setting their initial states
+      this.panels.forEach((panel, index) => {
+        if (panel !== this.openOnInit) {
+          // Set non-initialized accordion panels to hidden
+          this.panels[index].setAttribute('hidden', '');
+          let trigger = this.element.querySelector(`[${this.triggerAttribute}="${panel.dataset.accordionPanel}"]`);
+          // Set 'aria-expanded' for all non-initialized accordion triggers to 'false'
+          trigger.setAttribute('aria-expanded', 'false');
+        }
+      });
+    } else {
+      // Otherwise, keep all accordions closed to start
+      this.panels.forEach((panel, index) => {
+        // Set all of the accordion panels to hidden
+        this.panels[index].setAttribute('hidden', '');
+        let trigger = this.element.querySelector(`[${this.triggerAttribute}="${panel.dataset.accordionPanel}"]`);
+        // Set 'aria-expanded' for all of the accordion triggers to 'false'
+        trigger.setAttribute('aria-expanded', 'false');
+      });
+    }
+  }
```

Comment:
These `if/else` blocks are all duplication except for one check for the `openOnInit` option/panel. That most likely means these could be refactored into one `forEach` loop where you can do your logic.